### PR TITLE
Add the ability to manage (or not to manage) the Apache configuration file (typically httpd.conf)

### PR DESCRIPTION
**Pull request by bschonec**
I've got a whole bunch of existing installations and I don't [yet] want Puppet to manage my httpd.conf file.  This will give administrators the ability to leave the configuration file alone while they migrate to a fully puppetized configuration.

### CODE REVIEW

**Comment by HelenCampbell** [comment]
Hi,
Thanks for your pull request, however there are a few things this needs before being considered for merge. Your manage_conf_file parameter needs validation in the init.pp. You also would need a test to ensure it works and to then could you please squash your commits.

**Comment by bschonec** [comment]
As I'm very new at this I'm not sure how to submit a test.  Is it just a simple matter of modifying the ..tests/vhost.pp file and putting in an argument for $manage_conf_file?

**Comment by HelenCampbell** [comment]
You can find help for writing tests here: https://docs.puppetlabs.com/guides/module_guides/bgtm.html#step-three-module-testing

**Comment by bschonec** [comment]
I have added what I hope is a valid test.  I have never written a test before and I am no developer.  I read the documentation that you recommended but to be honest, I'm still in the dark as to the proper procedure.  I tried my best.

**Comment by bschonec** [comment]
I have added what I hope is a valid test.  I have never written a test before and I am no developer.  I read the documentation that you recommended but to be honest, I'm still in the dark as to the proper procedure.  I tried my best.

File: README.md [hunk 1]
```diff
@@ -2185,6 +2191,10 @@ This directive is equivalent to proxy_pass, but takes regular expressions, see [
 
 Specifies the resource identifiers for a rack configuration. The file paths specified are listed as rack application roots for [Phusion Passenger](http://www.modrails.com/documentation/Users%20guide%20Apache.html#_railsbaseuri_and_rackbaseuri) in the _rack.erb template. Defaults to 'undef'.
 
+#####`passenger_base_uris`
+
+Used to specify that the given URI is a Phusion Passenger-served application. The file paths specified are listed as passenger application roots for [Phusion Passenger](https://www.phusionpassenger.com/documentation/Users%20guide%20Apache.html#PassengerBaseURI) in the _passenger_base_uris.erb template. Defaults to 'undef'.
+
```
Comment (igalic) [inline README.md]: i'm confused about this docs addition.

**Comment by igalic** [comment]
mhm. from what i gather this seems to stem of your way of using git
maybe you wanna take a look at [GitHub flow](https://guides.github.com/introduction/flow/index.html)

**Comment by tphoney** [comment]
Hi @bschonec , thanks for the work you have put into this pull request. However due to inactivity no documentation / needs rebased / no tests / needs squashed this pull request is being closed. If you want to re-submit this PR with the necessary changes, we will gladly look at the new pull request.

**Pull request closed without merging**